### PR TITLE
[BEAM-242] Enable and fix checkstyle on Flink runner

### PR DESCRIPTION
**Pull request by jbonofre**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [X] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [X] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [X] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [X] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---
Enable checkstyle plugin in Flink runner and fix checkstyle issues.

### CODE REVIEW

**Comment by jbonofre** [comment]
R: @mxm 
R: @aljoscha 
R: @davorbonaci

**Comment by aljoscha** [comment]
Thanks for doing this @jbonofre!

I see that you are also fixing the examples. Almost all of these will probably go away so I think it doesn't make much sense to fix them (sorry for the extra work for you).

I'll read through the other changes tomorrow.

File: runners/flink/runner/src/main/java/org/apache/beam/runners/flink/TestFlinkPipelineRunner.java [hunk 1]
```diff
@@ -28,6 +28,9 @@
 
 import org.apache.flink.runtime.client.JobExecutionException;
 
+/**
+ * Test on Flink pipeline runner.
```
Comment (aljoscha) [inline runners/flink/runner/src/main/java/org/apache/beam/runners/flink/TestFlinkPipelineRunner.java]: "For running tests on {@link FlinkPipelineRunner}"

**Comment by aljoscha** [comment]
R: @davorbonaci I thought you might have something to say about Maven/Code Style

A few quick remarks from glancing at the code: We have checkstyle rules that enforce line length but we don't seem to have rules that specify how lines should be broken/chomped (IntelliJ uses this word for some setting). I'm seeing these snippets in the code:

```
FlinkPipelineOptions flinkOptions = PipelineOptionsValidator.validate(FlinkPipelineOptions
          .class, options);
```

```
StreamTransformTranslator<?> translator = FlinkStreamingTransformTranslators.getTranslator
            (transform);
```

```
private <T extends PTransform<?, ?>> void applyStreamingTransform(PTransform<?, ?> transform,
                                                                  TransformTreeNode node,
                                                                  StreamTransformTranslator<?>
                                                                  translator) {
```

```
private static final Map<Class<? extends PTransform>, FlinkStreamingPipelineTranslator
       .StreamTransformTranslator> TRANSLATORS = new HashMap<>();
```

```
public void translateNode(Create.Values<OutputT> transform, FlinkStreamingTranslationContext
         context) {
```

My feeling says that they should look something like this:

```
FlinkPipelineOptions flinkOptions =
    PipelineOptionsValidator.validate(FlinkPipelineOptions.class, options);
```

```
StreamTransformTranslator<?> translator =
    FlinkStreamingTransformTranslators.getTranslator(transform);
```

```
private <T extends PTransform<?, ?>> void applyStreamingTransform(
    PTransform<?, ?> transform,
    TransformTreeNode node,
    StreamTransformTranslator<?> translator) {
```

```
private static final Map<
    Class<? extends PTransform>,
    FlinkStreamingPipelineTranslator.StreamTransformTranslator> TRANSLATORS =
        new HashMap<>();
```
(about this one, with all the generics, I'm not sure... 😅) 

```
public void translateNode(
    Create.Values<OutputT> transform,
    FlinkStreamingTranslationContext context) {
```

In most of the the new changes things are done like this (thanks again @jbonofre), but if we don't enforce it the code can become very "strange looking".

File: runners/flink/examples/src/main/java/org/apache/beam/runners/flink/examples/streaming/KafkaWindowedWordCountExample.java [hunk 2]
```diff
@@ -134,7 +150,8 @@ public static void main(String[] args) {
     PCollection<String> words = pipeline
         .apply(Read.named("StreamingWordCount").from(UnboundedFlinkSource.of(kafkaConsumer)))
         .apply(ParDo.of(new ExtractWordsFn()))
-        .apply(Window.<String>into(FixedWindows.of(Duration.standardSeconds(options.getWindowSize())))
+        .apply(Window.<String>into(FixedWindows.of(Duration.standardSeconds(options.getWindowSize
+            ())))
```
Comment (mxm) [inline runners/flink/examples/src/main/java/org/apache/beam/runners/flink/examples/streaming/KafkaWindowedWordCountExample.java]:
```java
.apply(Window.<String>into(FixedWindows.of(Duration.standardSeconds(
    options.getWindowSize())))
```

File: runners/flink/examples/src/main/java/org/apache/beam/runners/flink/examples/streaming/WindowedWordCount.java [hunk 3]
```diff
@@ -62,7 +62,8 @@
   static class FormatAsStringFn extends DoFn<KV<String, Long>, String> {
     @Override
     public void processElement(ProcessContext c) {
-      String row = c.element().getKey() + " - " + c.element().getValue() + " @ " + c.timestamp().toString();
+      String row = c.element().getKey() + " - " + c.element().getValue() + " @ " + c.timestamp()
+          .toString();
```
Comment (mxm) [inline runners/flink/examples/src/main/java/org/apache/beam/runners/flink/examples/streaming/WindowedWordCount.java]: I'd like to see `c.timestamp().toString()` in one line.

File: runners/flink/examples/src/main/java/org/apache/beam/runners/flink/examples/streaming/WindowedWordCount.java [hunk 4]
```diff
@@ -104,7 +109,8 @@ public void processElement(ProcessContext c) {
   }
 
   public static void main(String[] args) throws IOException {
-    StreamingWordCountOptions options = PipelineOptionsFactory.fromArgs(args).withValidation().as(StreamingWordCountOptions.class);
+    StreamingWordCountOptions options = PipelineOptionsFactory.fromArgs(args).withValidation().as
+        (StreamingWordCountOptions.class);
```
Comment (mxm) [inline runners/flink/examples/src/main/java/org/apache/beam/runners/flink/examples/streaming/WindowedWordCount.java]: `as(..)` should be in one line.

File: runners/flink/examples/src/main/java/org/apache/beam/runners/flink/examples/streaming/KafkaWindowedWordCountExample.java [hunk 5]
```diff
@@ -109,15 +123,17 @@ public void processElement(ProcessContext c) {
...
     options.setJobName("KafkaExample - WindowSize: " + options.getWindowSize() + " seconds");
     options.setStreaming(true);
     options.setCheckpointingInterval(1000L);
     options.setNumberOfExecutionRetries(5);
     options.setExecutionRetryDelay(3000L);
     options.setRunner(FlinkPipelineRunner.class);
 
-    System.out.println(options.getKafkaTopic() +" "+ options.getZookeeper() +" "+ options.getBroker() +" "+ options.getGroup() );
+    System.out.println(options.getKafkaTopic() + " " + options.getZookeeper() + " " + options
+        .getBroker() + " " + options.getGroup());
```
Comment (mxm) [inline runners/flink/examples/src/main/java/org/apache/beam/runners/flink/examples/streaming/KafkaWindowedWordCountExample.java]: I would move the `options` to this line as well.

File: runners/flink/runner/src/main/java/org/apache/beam/runners/flink/TestFlinkPipelineRunner.java (hunk 1 above)
Comment (jbonofre) [inline runners/flink/runner/src/main/java/org/apache/beam/runners/flink/TestFlinkPipelineRunner.java]: OK, let me update ! Thanks @aljoscha

File: runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/FlinkGroupAlsoByWindowWrapper.java [hunk 6]
```diff
@@ -80,47 +81,53 @@
...
+ * fire (see {@link FlinkGroupAlsoByWindowWrapper#processWatermark(Watermark)}).
+ * These are deregistered from the {@link FlinkGroupAlsoByWindowWrapper#activeTimers}
+ * list, and are fed into the {@link org.apache.beam.sdk.util.GroupAlsoByWindowsDoFn}
+ * for further processing.
+ * </p>
  */
-public class FlinkGroupAlsoByWindowWrapper<K, VIN, VACC, VOUT>
-    extends AbstractStreamOperator<WindowedValue<KV<K, VOUT>>>
-    implements OneInputStreamOperator<WindowedValue<KV<K, VIN>>, WindowedValue<KV<K, VOUT>>> {
+public class FlinkGroupAlsoByWindowWrapper<K, InputT, AccumT, OutputT>
```
Comment (mxm) [inline runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/FlinkGroupAlsoByWindowWrapper.java]: Are generic type names also mandated by the style?

**Comment by jbonofre** [comment]
FYI, I used IntelliJ line wrapping.

File: runners/flink/examples/src/main/java/org/apache/beam/runners/flink/examples/streaming/KafkaWindowedWordCountExample.java (hunk 2 above)
Comment (jbonofre) [inline runners/flink/examples/src/main/java/org/apache/beam/runners/flink/examples/streaming/KafkaWindowedWordCountExample.java]: OK, let me update.

File: runners/flink/examples/src/main/java/org/apache/beam/runners/flink/examples/streaming/WindowedWordCount.java (hunk 4 above)
Comment (jbonofre) [inline runners/flink/examples/src/main/java/org/apache/beam/runners/flink/examples/streaming/WindowedWordCount.java]: OK let me update (I used IntelliJ line wrapping).

File: runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/FlinkGroupAlsoByWindowWrapper.java (hunk 6 above)
Comment (jbonofre) [inline runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/FlinkGroupAlsoByWindowWrapper.java]: Yes, our checkstyle rules forces regex on type names.

File: runners/flink/examples/src/main/java/org/apache/beam/runners/flink/examples/streaming/KafkaWindowedWordCountExample.java (hunk 5 above)
Comment (jbonofre) [inline runners/flink/examples/src/main/java/org/apache/beam/runners/flink/examples/streaming/KafkaWindowedWordCountExample.java]: OK, I will update.

File: runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/UnboundedFlinkSource.java [hunk 7]
```diff
@@ -82,11 +88,13 @@ public void setCoder(Coder<T> coder) {
 
   /**
    * Creates a new unbounded source from a Flink source.
+   *
    * @param flinkSource The Flink source function
-   * @param <T> The type that the source function produces.
+   * @param <T>         The type that the source function produces.
```
Comment (mxm) [inline runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/UnboundedFlinkSource.java]: Space

File: runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/UnboundedSocketSource.java [hunk 8]
```diff
@@ -92,27 +95,30 @@ public long getDelayBetweenRetries() {
...
     // In our case we do not need a coder.
     return null;
   }
 
   @Override
   public void validate() {
     checkArgument(port > 0 && port < 65536, "port is out of range");
-    checkArgument(maxNumRetries >= -1, "maxNumRetries must be zero or larger (num retries), or -1 (infinite retries)");
+    checkArgument(maxNumRetries >= -1, "maxNumRetries must be zero or larger (num retries), or -1"
+        + " (infinite retries)");
```
Comment (mxm) [inline runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/UnboundedSocketSource.java]: For reabability, the `-1` should also go in this line.

File: runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/UnboundedSocketSource.java [hunk 9]
```diff
@@ -141,7 +151,8 @@ public UnboundedSocketReader(UnboundedSocketSource source) {
 
     private void openConnection() throws IOException {
       this.socket = new Socket();
-      this.socket.connect(new InetSocketAddress(this.source.getHostname(), this.source.getPort()), CONNECTION_TIMEOUT_TIME);
+      this.socket.connect(new InetSocketAddress(this.source.getHostname(), this.source.getPort())
+          , CONNECTION_TIMEOUT_TIME);
```
Comment (mxm) [inline runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/UnboundedSocketSource.java]: Space

File: runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/UnboundedSocketSource.java [hunk 10]
```diff
@@ -152,11 +163,14 @@ public boolean start() throws IOException {
...
-          LOG.info("Connected to server socket " + this.source.getHostname() + ':' + this.source.getPort());
+          LOG.info("Connected to server socket " + this.source.getHostname() + ':' + this.source
+              .getPort());
 
           return advance();
         } catch (IOException e) {
-          LOG.info("Lost connection to server socket " + this.source.getHostname() + ':' + this.source.getPort() + ". Retrying in " + this.source.getDelayBetweenRetries() + " msecs...");
+          LOG.info("Lost connection to server socket " + this.source.getHostname() + ':' + this
+              .source.getPort() + ". Retrying in " + this.source.getDelayBetweenRetries() + " "
+              + "msecs...");
```
Comment (mxm) [inline runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/UnboundedSocketSource.java]:
```java
 .source.getPort() + ". Retrying in " +
    this.source.getDelayBetweenRetries() + " msecs...");

File: runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/UnboundedSocketSource.java [hunk 11]
```diff
@@ -214,7 +229,8 @@ public void close() throws IOException {
       this.reader.close();
       this.socket.close();
       this.isRunning = false;
-      LOG.info("Closed connection to server socket at " + this.source.getHostname() + ":" + this.source.getPort() + ".");
+      LOG.info("Closed connection to server socket at " + this.source.getHostname() + ":" + this
+          .source.getPort() + ".");
```
Comment (mxm) [inline runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/UnboundedSocketSource.java]: `this` should go in the next line.

File: runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/state/StateCheckpointReader.java [hunk 12]
```diff
@@ -80,7 +83,8 @@ public Instant getTimestamp() throws IOException {
     byte[] serData = new byte[size];
     int bytesRead = input.read(serData);
     if (bytesRead != size) {
-      throw new RuntimeException("Error while deserializing checkpoint. Not enough bytes in the input stream.");
+      throw new RuntimeException("Error while deserializing checkpoint. Not enough bytes in the "
+          + "input stream.");
```
Comment (mxm) [inline runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/state/StateCheckpointReader.java]: I think this is clearer:

```java
throw new RuntimeException("Error while deserializing checkpoint."
    + " Not enough bytes in the input stream.");
```

File: runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/state/StateCheckpointWriter.java [hunk 13]
```diff
@@ -27,19 +27,24 @@
...
     return new StateCheckpointWriter(output);
   }
 
   private StateCheckpointWriter(AbstractStateBackend.CheckpointStateOutputView output) {
     this.output = output;
   }
 
-  /////////      Creating the serialized versions of the different types of state held by dataflow      ///////
+  /////////      Creating the serialized versions of the different types of state held by
+  // beam      ///////
```
Comment (mxm) [inline runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/state/StateCheckpointWriter.java]: Formatting is off.

File: runners/flink/runner/src/test/java/org/apache/beam/runners/flink/PipelineOptionsTest.java [hunk 14]
```diff
@@ -36,13 +37,14 @@
...
     String getTestOption();
+
     void setTestOption(String value);
   }
 
   private static MyOptions options;
   private static SerializedPipelineOptions serializedOptions;
 
-  private final static String[] args = new String[]{"--testOption=nothing"};
+  private static String[] args = new String[]{"--testOption=nothing"};
```
Comment (mxm) [inline runners/flink/runner/src/test/java/org/apache/beam/runners/flink/PipelineOptionsTest.java]: Why remove this `final` modifier here?

File: runners/flink/runner/src/test/java/org/apache/beam/runners/flink/streaming/UnboundedSourceWrapperTest.java [hunk 15]
```diff
@@ -58,14 +58,14 @@
    */
   @Test
   public void testWithOneReader() throws Exception {
-    final int NUM_ELEMENTS = 20;
+    final int numElements = 20;
```
Comment (mxm) [inline runners/flink/runner/src/test/java/org/apache/beam/runners/flink/streaming/UnboundedSourceWrapperTest.java]: https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names

**Comment by mxm** [comment]
Thanks for the PR, @jbonofre! I agree with @aljoscha but I think line wrapping should always be followed by +4 indention according to the Google style: https://google.github.io/styleguide/javaguide.html#s4.5.2-line-wrapping-indent

>```java
private static final Map<
    Class<? extends PTransform>,
    FlinkStreamingPipelineTranslator.StreamTransformTranslator> TRANSLATORS =
        new HashMap<>();
```

Shouldn't this be:

```java
private static final Map<
    Class<? extends PTransform>,
        FlinkStreamingPipelineTranslator.StreamTransformTranslator> TRANSLATORS = 
            new HashMap<>();
```

Unfortunately, the style is quite extensive and not fully checked by the Checkstyle rules.

In addition, I've made some more comments concerning sensible wrapping of lines, generic type name changes, and final/constants capitalization.

File: runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/UnboundedSocketSource.java (hunk 8 above)
Comment (bjchambers) [inline runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/UnboundedSocketSource.java]: Another option would be to move the entire string to the next line and keep it together:

```
checkArgument(maxNumRetries >= -1,
    "maxNumRetries must be zero or larger (num retries), or -1 (infinite retries)");
```

File: runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/FlinkGroupByKeyWrapper.java [hunk 16]
```diff
@@ -64,10 +66,12 @@ public K getKey(WindowedValue<KV<K, V>> value) throws Exception {
...
 
-  // special type to return as key for null key
+  /**
+   * Special type to return  as key for null key.
+   */
   public static class VoidValue {
     private VoidValue() {}
 
-    public static VoidValue INSTANCE = new VoidValue();
+    public static VoidValue instance = new VoidValue();
```
Comment (bjchambers) [inline runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/FlinkGroupByKeyWrapper.java]: I think the style guide says that only constants are capitalized, so `static final VoidValue INSTANCE`. It seems like making this final might be reasonable?

**Comment by davorbonaci** [comment]
(Not release blocking, IMO.)

**Comment by davorbonaci** [comment]
Update: I'd try to get javadoc fixes out, if possible, before the first release. Checkstyle part can be addressed later.

**Comment by jbonofre** [comment]
FYI, javadoc fails on flink runner using the release profile: mvn clean install -Prelease.

I'm extracting javadoc commits from this PR in another PR.

**Comment by kennknowles** [comment]
BTW this formatter is quite good: https://github.com/google/styleguide/blob/gh-pages/eclipse-java-google-style.xml

**Comment by kennknowles** [comment]
Here is what I meant to link: https://github.com/google/google-java-format

**Comment by mxm** [comment]
+1 maybe we can check out if the Google Java Format tool does a better job than the IntelliJ line wrapping.

**Comment by jbonofre** [comment]
I'm closing this PR for now.

**Pull request closed without merging**